freshmaker/pyxis.py

```diff
@@ -215,6 +215,129 @@ def get_images_by_digests(self, digests):
+                            reg_repo_pair[0], reg_repo_pair[1])
+                continue
+            tags = repo.get('auto_rebuild_tags')
+            # If the repository doesn't have 'auto_rebuild_tags', don't proceed with it
+            if tags:
+                reg_repo_info[reg_repo_pair]['tags'] = tags
+            else:
+                del reg_repo_info[reg_repo_pair]
+
+    def _filter_auto_rebuild_repos(self, reg_repo_info):
+        """
+        For every pair in mapping get history of every auto_rebuild tag with
+        filter of all nvrs from that registry/repo pair.
+
+        :param dict reg_repo_info: map of pairs (registry, repository) to a
+            dict containing nvrs of bundle images from that repo and
+            auto_rebuild tags of that repo
+        :return: pairs of (registry, repository) of every repo with images ever
+            published to a repo with auto_rebuild tag
+        :rtype set(tuple(str, str)):
+        """
+        params = {'include': 'data.brew.build'}
+        ret_repos = set()
+        for (reg, repo), info in reg_repo_info.items():
+            if not info.get('nvrs'):
+                continue
+            fltr = f"brew.build=in=({','.join(info['nvrs'])})"
+            params["filter"] = fltr
+            for tag in info.get('tags', []):
+                tag_history = \
+                    self._pagination(f"repositories/registry/{reg}/"
+                                     f"repository/{repo}/tag/{tag}", params)
+                # If there is at least one record with nvr(brew.build) of the
+                # image, it means that image was published to a repo with
+                # auto_rebuild tag
+                if tag_history:
+                    ret_repos.add((reg, repo))
+                    break
+        return ret_repos
+
+    def get_auto_rebuild_taged_images(self, bundle_images):
+        """
+        Determine which bundle images are part of container repository and tag
+        that has auto-rebuilds enabled
+
+        :param list bundle_images: Images of operator bundles that should
+            be filtered
+        :return: Images that where published to repository with auto_rebuild tag
+        :rtype: list
+        """
+        reg_repo_info = {}
+        ret_images = []
+        for bundle_image in bundle_images:
+            bundle_nvr = bundle_image.get('brew', {}).get('build')
+            if not bundle_nvr:
+                log.warning("One of bundle images doesn't have brew.build")
+                continue
+            if not bundle_image.get('repositories'):
+                log.warning("Bundle image %s doesn't have repositories set",
+                            bundle_nvr)
+                continue
+            # construct mapping of (registry, repository) -> {'nvrs': {bundles_nvrs}}
+            for repo in bundle_image.get('repositories'):
+                reg_repo = (repo.get('registry'), repo.get('repository'))
```

Comment:
Can `registry` or `repository` not be set? If so, that seems like something we should ignore. If it's always set, then you should just access the keys directly to insinuate that it's always set and no handling is required for when it is not set.